stellarphot/photometry/photometry.py

```diff
@@ -255,8 +255,8 @@ def single_image_photometry(ccd_image, sourcelist, camera, observatory_location,
                "SKIPPING THIS IMAGE!")
         return None, None
 
-    # Set high pixels to NaN
-    ccd_image.data[ccd_image.data > max_adu] = np.nan
+    # Set high pixels to NaN (make sure ccd_image.data is a float array first)
```

Comment:
I think you still need a line here making the data float, it just doesn't need a type check:

```suggestion
    # Set high pixels to NaN (make sure ccd_image.data is a float array first)
    ccd_image.data = ccd_image.data.astype(float)
```

Reply:
Didn't I cast it to float during the NaN replacement? (On iPhone, can't see code)

Reply:
You cast it as float for the comparison (inside the brackets) but the data itself is still int I think

Reply:
I think I fixed it now.